var/spack/repos/jcsda-emc-bundles/packages/base-env/package.py

```diff
@@ -27,12 +30,17 @@ class BaseEnv(BundlePackage):
     depends_on("curl", type="run")
 
     # I/O
-    depends_on("zlib", type="run")
-    depends_on("hdf5", type="run")
-    depends_on("netcdf-c", type="run")
-    depends_on("netcdf-fortran", type="run")
-    depends_on("parallel-netcdf", type="run")
-    depends_on("parallelio", type="run")
+    depends_on("zlib+shared", type="run", when="+shared")
+    depends_on("zlib~shared", type="run", when="~shared")
+    depends_on("hdf5+hl+mpi+shared", type="run", when="+shared")
+    depends_on("hdf5+hl+mpi~shared", type="run", when="~shared")
+    depends_on("netcdf-c~parallel-netcdf+v2+mpi+shared", type="run", when="+shared")
+    depends_on("netcdf-c~parallel-netcdf+v2+mpi~shared", type="run", when="~shared")
+    depends_on("netcdf-fortran+shared", type="run", when="+shared")
+    depends_on("netcdf-fortran~shared", type="run", when="~shared")
+    depends_on("parallel-netcdf+shared", type="run", when="+shared")
+    depends_on("parallel-netcdf~shared", type="run", when="~shared")
+    depends_on("parallelio+fortran~pnetcdf~shared", type="run")
```

Comment:
This line needs a switch for `+shared`. We should fix the logic in the ufs-weather-model build and remove the explicit request for `STATIC` there, just take whatever spack-stack has built.